src/include/daos_obj_class.h

```diff
@@ -240,149 +240,423 @@ enum {
+	OC_EC_L8P1G1K,
+	OC_EC_L8P1G2K,
+	OC_EC_L8P1G4K,
+	OC_EC_L8P1G8K,
+	OC_EC_L8P1GX,
+
+	/** EC 8+2, small cell size, object classes */
+	OC_EC_S8P2G1    = 580,
+	OC_EC_S8P2G2,
+	OC_EC_S8P2G4,
+	OC_EC_S8P2G8,
+	OC_EC_S8P2G16,
+	OC_EC_S8P2G32,
+	OC_EC_S8P2G64,
+	OC_EC_S8P2G128,
+	OC_EC_S8P2G256,
+	OC_EC_S8P2G512,
+	OC_EC_S8P2G1K,
+	OC_EC_S8P2G2K,
+	OC_EC_S8P2G4K,
+	OC_EC_S8P2G8K,
+	OC_EC_S8P2GX,
+
+	/** EC 8+2, medium cell size, object classes */
+	OC_EC_M8P2G1    = 600,
+	OC_EC_M8P2G2,
+	OC_EC_M8P2G4,
+	OC_EC_M8P2G8,
+	OC_EC_M8P2G16,
+	OC_EC_M8P2G32,
+	OC_EC_M8P2G64,
+	OC_EC_M8P2G128,
+	OC_EC_M8P2G256,
+	OC_EC_M8P2G512,
+	OC_EC_M8P2G1K,
+	OC_EC_M8P2G2K,
+	OC_EC_M8P2G4K,
+	OC_EC_M8P2G8K,
+	OC_EC_M8P2GX,
+
+	/** EC 8+2, large cell size, object classes */
+	OC_EC_L8P2G1    = 620,
+	OC_EC_L8P2G2,
+	OC_EC_L8P2G4,
+	OC_EC_L8P2G8,
+	OC_EC_L8P2G16,
+	OC_EC_L8P2G32,
+	OC_EC_L8P2G64,
+	OC_EC_L8P2G128,
+	OC_EC_L8P2G256,
+	OC_EC_L8P2G512,
+	OC_EC_L8P2G1K,
+	OC_EC_L8P2G2K,
+	OC_EC_L8P2G4K,
+	OC_EC_L8P2G8K,
+	OC_EC_L8P2GX,
+
+	/** EC 16+1, small cell size, object classes */
+	OC_EC_S16P1G1    = 640,
+	OC_EC_S16P1G2,
+	OC_EC_S16P1G4,
+	OC_EC_S16P1G8,
+	OC_EC_S16P1G16,
+	OC_EC_S16P1G32,
+	OC_EC_S16P1G64,
+	OC_EC_S16P1G128,
+	OC_EC_S16P1G256,
+	OC_EC_S16P1G512,
+	OC_EC_S16P1G1K,
+	OC_EC_S16P1G2K,
+	OC_EC_S16P1G4K,
+	OC_EC_S16P1G8K,
+	OC_EC_S16P1GX,
+
+	/** EC 16+1, medium cell size, object classes */
+	OC_EC_M16P1G1    = 660,
+	OC_EC_M16P1G2,
+	OC_EC_M16P1G4,
+	OC_EC_M16P1G8,
+	OC_EC_M16P1G16,
+	OC_EC_M16P1G32,
+	OC_EC_M16P1G64,
+	OC_EC_M16P1G128,
+	OC_EC_M16P1G256,
+	OC_EC_M16P1G512,
+	OC_EC_M16P1G1K,
+	OC_EC_M16P1G2K,
+	OC_EC_M16P1G4K,
+	OC_EC_M16P1G8K,
+	OC_EC_M16P1GX,
+
+	/** EC 16+1, large cell size, object classes */
+	OC_EC_L16P1G1    = 680,
+	OC_EC_L16P1G2,
+	OC_EC_L16P1G4,
+	OC_EC_L16P1G8,
+	OC_EC_L16P1G16,
+	OC_EC_L16P1G32,
+	OC_EC_L16P1G64,
+	OC_EC_L16P1G128,
+	OC_EC_L16P1G256,
+	OC_EC_L16P1G512,
+	OC_EC_L16P1G1K,
+	OC_EC_L16P1G2K,
+	OC_EC_L16P1G4K,
+	OC_EC_L16P1G8K,
+	OC_EC_L16P1GX,
+
+	/** EC 16+2, small cell size, object classes */
+	OC_EC_S16P2G1    = 700,
+	OC_EC_S16P2G2,
+	OC_EC_S16P2G4,
+	OC_EC_S16P2G8,
+	OC_EC_S16P2G16,
+	OC_EC_S16P2G32,
+	OC_EC_S16P2G64,
+	OC_EC_S16P2G128,
+	OC_EC_S16P2G256,
+	OC_EC_S16P2G512,
+	OC_EC_S16P2G1K,
+	OC_EC_S16P2G2K,
+	OC_EC_S16P2G4K,
+	OC_EC_S16P2G8K,
+	OC_EC_S16P2GX,
+
+	/** EC 16+2, small cell size, object classes */
+	OC_EC_M16P2G1    = 720,
+	OC_EC_M16P2G2,
+	OC_EC_M16P2G4,
+	OC_EC_M16P2G8,
+	OC_EC_M16P2G16,
+	OC_EC_M16P2G32,
+	OC_EC_M16P2G64,
+	OC_EC_M16P2G128,
+	OC_EC_M16P2G256,
+	OC_EC_M16P2G512,
+	OC_EC_M16P2G1K,
+	OC_EC_M16P2G2K,
+	OC_EC_M16P2G4K,
+	OC_EC_M16P2G8K,
+	OC_EC_M16P2GX,
+
+	/** EC 16+2, small cell size, object classes */
+	OC_EC_L16P2G1    = 740,
+	OC_EC_L16P2G2,
+	OC_EC_L16P2G4,
+	OC_EC_L16P2G8,
+	OC_EC_L16P2G16,
+	OC_EC_L16P2G32,
+	OC_EC_L16P2G64,
+	OC_EC_L16P2G128,
+	OC_EC_L16P2G256,
+	OC_EC_L16P2G512,
+	OC_EC_L16P2G1K,
+	OC_EC_L16P2G2K,
+	OC_EC_L16P2G4K,
+	OC_EC_L16P2G8K,
+	OC_EC_L16P2GX,
```

Comment:
As we discussed in the EC meeting, I think we should move these classes to src/include/daos/object.h
and make the public one to be "medium":
```
#define OC_EC_M16P2G1         OC_EC_16P2G1
#define OC_EC_M16P2G2         OC_EC_16P2G2
```
In addition, we have to use binary search instead of scanning array now, otherwise it's going to be too slow. I can work out a patch for this.